src/main/java/org/haffson/adventofcode/days/day06/Coordinate.java

```diff
@@ -0,0 +1,358 @@
+package org.haffson.adventofcode.days.day06;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Helper record for day 6.
+ * Is used to work better with the string inputs.
+ * Coordinates have an x and y value.
+ * Also, you can compare two coordinates with each other.
+ */
+public record Coordinate(int xCoordinate, int yCoordinate) {
+
+    @Nonnull
+    public static Coordinate of(@Nonnull final String input) {
+        final String[] split1 = input.split(",");
+        final String[] split2 = split1[1].split(" ");
+
+        final int xCoordinate = Integer.parseInt(split1[0]);
+        final int yCoordinate = Integer.parseInt(split2[1]);
+
+        return new Coordinate(xCoordinate, yCoordinate);
+    }
+
+    /**
+     * Helper method for day 6.
+     * Gives a list<Coordinate> of a string-list.
+     *
+     * @return List<Coordinate>
+     */
+    @Nonnull
+    public static List<Coordinate> of(@Nonnull final List<String> stringList) {
+        return stringList.stream().map(Coordinate::of).toList();
+    }
+
+    /**
+     * Helper method for day 6.
+     * Compares the distance of two coordinates.
+     *
+     * @return int of how many fields are between two coordinates.
+     */
+    public static int coordinateFieldDistance(@Nonnull final Coordinate coordinateToCompare,
+                                              @Nonnull final Coordinate comparedCoordinate) {
+        int xDistance = Math.abs(coordinateToCompare.xCoordinate - comparedCoordinate.xCoordinate);
+        int yDistance = Math.abs(coordinateToCompare.yCoordinate - comparedCoordinate.yCoordinate);
+
+        return xDistance + yDistance;
+    }
+
+    /**
+     * Helper record for day 6.
+     * Uses a List<Coordinates> to get a coordinate system.
+     * Has some helper methods to calculate areas of coordinates.
+     */
+    public record CoordinateAreas(@Nonnull List<Coordinate> coordinates, @Nonnull String[][] coordinateSystem) {
+
+        public CoordinateAreas(@Nonnull List<Coordinate> coordinates) {
+            this(coordinates, createCoordinateSystem(coordinates));
+        }
+
+        /**
+         * Helper method for day 6.
+         * Creates a coordinate system.
+         * The length and height depends on the max values of the List<Coordinate>
+         *
+         * @return String[][] coordinate system.
+         */
+        @Nonnull
+        private static String[][] createCoordinateSystem(@Nonnull final List<Coordinate> coordinates) {
+            int maxCoordinateX = coordinates.stream()
+                    .max(Comparator.comparingInt(c -> c.xCoordinate))
```

Comment:
Gewöhne dir bitte nicht an, solche kurzen Parameternamen zu nutzen. Das mag jetzt im Moment eindeutig wirken, aber sobald etwas geändert wurde, kann diese Eindeutigkeit verloren gehen. Im Frontend, wo TypeScript oder JavaScript bekanntlich sehr dynamisch sind, wirst du nicht die Typesafety von Java haben. Und genau da ist es uns bei NLP schon auf die Füße gefallen. 

```suggestion
                    .max(Comparator.comparingInt(coordinate -> coordinate.xCoordinate))
```

Reply:
Das müsste übrigens auch gehen:

```suggestion
                    .max(Comparator.comparingInt(Coordinate::xCoordinate))
```

Reply:
Wenn es um die Länge der Variable geht, dann kann man in diesem Fall auch wieder gut kürzen:

    coordinate.x

Dass X zu einer Koordinate gehört wird dann ja direkt klar.

Reply:
Klingt logisch. Tatsächlich hatte ich etwas ähnliches schon verbessert bei "claim". Werde da noch stärker drauf achten.
Die andere Schreibweise funktioniert übrigens. Bin leicht verwundert, weil ich diese Schreibweise eigentlich zu Beginn genutzt hatte, aber etwas nicht funktioniert hatte - da lag der Fehler wohl an anderer Stelle